test/tools/ossfuzz/protoToAbiV2.h

```diff
@@ -19,84 +19,118 @@
+ *			// Check if calls to coder_public meet expectations for correctly/incorrectly encoded data.
+ *			if (returnVal != 0)
+ *				return returnVal;
  *
- *      /// Compares two dynamically sized bytes arrays for equality.
- *      function bytesCompare(bytes memory a, bytes memory b) internal pure returns (bool) {
- *          ...
- *      }
+ *			returnVal = checkEncodedCall(this.coder_external.selector, argumentEncoding, <invalidLengthFuzz>);
+ *			// Check if calls to coder_external meet expectations for correctly/incorrectly encoded data.
+ *			// Offset return value to distinguish between failures originating from coder_public and coder_external.
+ *			if (returnVal != 0)
+ *				return uint(200000) + returnVal;
+ *			// Return zero if all checks pass.
+ *			return 0;
+ *		}
  *
- *      // Public function that is called by test() function. Accepts one or more arguments and returns
- *      // a uint value (zero if abi en/decoding was successful, non-zero otherwise)
- *      function coder_public(string memory c_0, bytes memory c_1) public pure returns (uint) {
- *              if (!bytesCompare(bytes(c_0), "044852b2a670ade5407e78fb2863c51de9fcb96542a07186fe3aeda6bb8a116d"))
- *	    	        return 1;
- *              if (!bytesCompare(c_1, "1"))
- *                  return 2;
- *              return 0;
- *	    }
+ *		/// Accepts function selector, correct argument encoding, and an invalid encoding length as input.
+ *		/// Returns a non-zero value if either call with correct encoding fails or call with incorrect encoding
+ *		/// succeeds. Returns zero if both calls meet expectation.
+ *  	function checkEncodedCall(bytes4 funcSelector, bytes memory argumentEncoding, uint invalidLengthFuzz)
+ *  		public returns (uint) {
+ *  		...
+ *  	}
  *
- *      // External function that is called by test() function. Accepts one or more arguments and returns
- *      // a uint value (zero if abi en/decoding was successful, non-zero otherwise)
- *      function coder_external(string calldata c_0, bytes calldata c_1) external pure returns (uint) {
- *              if (!bytesCompare(bytes(c_0), "044852b2a670ade5407e78fb2863c51de9fcb96542a07186fe3aeda6bb8a116d"))
- *  		        return 1;
- *              if (!bytesCompare(c_1, "1"))
- *                  return 2;
- *              return 0;
- *	    }
- * }
+ *		/// Accepts function selector, correct argument encoding, and length of invalid encoding and returns
+ *		/// the correct and incorrect abi encoding for calling the function specified by the function selector.
+ *  	function createEncoding(bytes4 funcSelector, bytes memory argumentEncoding, uint invalidLengthFuzz)
+ *  		internal pure returns (bytes memory, bytes memory) {
+ *  		...
+ *  	}
+ *
+ *		/// Compares two dynamically sized bytes arrays for equality.
+ *		function bytesCompare(bytes memory a, bytes memory b) internal pure returns (bool) {
```

Comment:
I almost approved, but, to be picky: this looks differently indented than the function above and below ;-).

Reply:
Ah, you are right. The indenting there is actually quite messed up.